app/controllers/samples_controller.rb

```diff
@@ -30,8 +30,10 @@ class SamplesController < ApplicationController
                    :dimensions, :all, :show_sample_names, :cli_user_instructions, :metadata_fields, :samples_going_public,
                    :search_suggestions, :stats, :upload, :validate_sample_files].freeze
 
-  before_action :authenticate_user!, except: [:create, :update, :bulk_upload, :bulk_upload_with_metadata]
-  before_action :authenticate_user_from_token!, only: [:create, :update, :bulk_upload, :bulk_upload_with_metadata]
+  # For API-like access
+  TOKEN_AUTH_ACTIONS = [:create, :update, :bulk_upload, :bulk_upload_with_metadata, :report_info].freeze
```

Comment:
nice

Reply:
Actually, no! I think `report_info` must not be in TOKEN_AUTH_ACTIONS. Will follow up offline